core/src/main/scala/cats/data/Const.scala

```diff
@@ -6,6 +6,7 @@ import cats.functor.Contravariant
 /**
  * [[Const]] is a phantom type, it does not contain a value of its second type parameter `B`
  * [[Const]] can be seen as a type level version of `Function.const[A, B]: A => B => A`
+ * B is set covariant to help type inference.
```

Comment:
Oh, I saw a discussion below, I think you forgot to make `B` covariant in `Const` definition